numba/targets/linalg.py

```diff
@@ -667,3 +688,300 @@ def inv(context, builder, sig, args):
+                # XXX Py_FatalError()?
+                raise RuntimeError("unable to execute xxpotrf()")
+            if info[0] > 0:
+                raise numpy.linalg.LinAlgError(
+                    "Matrix is not positive definite.")
+            elif info[0] < 0:
+                # Invalid parameter
+                raise RuntimeError(
+                    "cholesky(): Internal error: please report an issue")
+            # Zero out upper triangle, in F order
+            for col in range(n):
+                out[:col, col] = 0
+            return out
+
+        return cho_impl
+
+    @overload(numpy.linalg.eig)
+    def eig_impl(a):
+        ensure_lapack()
+
+        _check_linalg_matrix(a, "eig")
+
+        numba_ez_rgeev_sig = types.intc(types.char,  # kind
+                                        types.char,  # jobvl
+                                        types.char,  # jobvr
+                                        types.intp,  # n
+                                        types.CPointer(a.dtype),  # a
+                                        types.intp,  # lda
+                                        types.CPointer(a.dtype),  # wr
+                                        types.CPointer(a.dtype),  # wi
+                                        types.CPointer(a.dtype),  # vl
+                                        types.intp,  # ldvl
+                                        types.CPointer(a.dtype),  # vr
+                                        types.intp  # ldvr
+                                        )
+
+        numba_ez_rgeev = types.ExternalFunction("numba_ez_rgeev",
+                                                numba_ez_rgeev_sig)
+
+        numba_ez_cgeev_sig = types.intc(types.char,  # kind
+                                        types.char,  # jobvl
+                                        types.char,  # jobvr
+                                        types.intp,  # n
+                                        types.CPointer(a.dtype),  # a
+                                        types.intp,  # lda
+                                        types.CPointer(a.dtype),  # w
+                                        types.CPointer(a.dtype),  # vl
+                                        types.intp,  # ldvl
+                                        types.CPointer(a.dtype),  # vr
+                                        types.intp  # ldvr
+                                        )
+
+        numba_ez_cgeev = types.ExternalFunction("numba_ez_cgeev",
+                                                numba_ez_cgeev_sig)
+
+        kind = ord(get_blas_kind(a.dtype, "eig"))
+
+        JOBVL = ord('N')
+        JOBVR = ord('V')
+
+        F_layout = a.layout == 'F'
+
+        def real_eig_impl(a):
+            n = a.shape[-1]
+            if a.shape[-2] != n:
+                msg = "Last 2 dimensions of the array must be square."
+                raise numpy.linalg.LinAlgError(msg)
+
+            if not numpy.isfinite(a).all():
```

Comment:
That will allocate an intermediate array just for the `isfinite()` result. It would be better to write the loop by hand:
```
for v in numpy.nditer(a):
    if not numpy.isfinite(v.item()):
        raise LinAlgError("etc.")
```